crypto/fipsmodule/cpucap/cpu_aarch64_netbsd.c

```diff
@@ -0,0 +1,131 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/cpu.h>
+
+#if defined(OPENSSL_AARCH64) && defined(OPENSSL_NETBSD) && \
+    !defined(OPENSSL_STATIC_ARMCAP)
+
+#include <aarch64/armreg.h>
+#include <aarch64/cpu.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+
+#include <openssl/arm_arch.h>
+
+#include "internal.h"
+
+// Helper function to query a specific CPU's capabilities
+static int get_cpu_id(int cpu_num, struct aarch64_sysctl_cpu_id *id) {
+  char sysctl_name[64];
+  size_t len = sizeof(*id);
+
+  snprintf(sysctl_name, sizeof(sysctl_name), "machdep.cpu%d.cpu_id", cpu_num);
+
+  if (sysctlbyname(sysctl_name, id, &len, NULL, 0) < 0) {
+    return -1;
+  }
+
+  if (len != sizeof(*id)) {
+    return -1;
+  }
+
+  return 0;
+}
+
+void OPENSSL_cpuid_setup(void) {
+  struct aarch64_sysctl_cpu_id cpu_id;
+
+  // NetBSD's machdep.cpuN.cpu_id sysctl reads each core's ID registers
+  // directly, so it reflects that specific core's capabilities, not a
+  // system-wide minimum.
+
+  // Initialize with all features enabled (we'll AND them together)
+  uint64_t common_aa64isar0 = UINT64_MAX;
+  uint64_t common_aa64pfr0 = UINT64_MAX;
+  int found_cpu = 0;
+
+  // Query up to 256 CPUs (arbitrary but reasonable upper limit)
+  for (int cpu_num = 0; cpu_num < 256; cpu_num++) {
+    if (get_cpu_id(cpu_num, &cpu_id) < 0) {
+      // Failed to read this CPU - either it doesn't exist or is offline
+      if (found_cpu > 0) {
+        // We've already found at least one CPU, so assume we've enumerated all
+        break;
```

Comment:
The assumption is that the remaining CPUs can't be probed or it's some optimisation through simplification?